### DISCUSSION TITLE
Best approach to sort table rows via stock TableWidget

### DISCUSSION THREAD

**Discussion by 38dh91** [AllenDang/giu#743] [Q&A] [answered]
Good day!

I am in doubt how can I sort rows on the stock TableWidget (TableColumnWidget) in intended way? As I understand, there is no way to provide event handling on header's cells. As a workaround I suppose to use regular TableRowWidget, but may be there is more elegant way to solve it.

Thank you in advance!

**Comment by 38dh91** [comment]
Solved with cimgui-go

**Reply by SolenOchHavet** [reply]
How did you solve it with cimgui-go? I'm stuck with the same issue :/

**Reply by AllenDang** [reply]
Don't use header's cells, use a row and fix it as header and you could got any events you need.

**Reply by 38dh91** [reply, marked as answer]
@SolenOchHavet something like this:
```
if imgui.BeginTableV("##table", 2, getTableFlags(), vec2(0, 0), 0) {
  imgui.TableSetupScrollFreeze(0, 1)
  imgui.TableSetupColumnV("ID##table_col_id", imgui.TableColumnFlagsNoHide|imgui.TableColumnFlagsWidthFixed, imgui.CalcTextSize("00000000").X, 0)
  imgui.TableSetupColumnV("Text##table_col_text, 0, 0, 1)
  imgui.TableHeadersRow()
  
  // get table sort specs
  tableSorting := imgui.TableGetSortSpecs()
  if tableSorting.SpecsCount() != 0 {
    switch tableSorting.Specs().ColumnUserID() {
      case 1:
        logger.Info("%v", tableSorting.Specs().SortDirection())
    }
  }
  ...
}
```

**Reply by SolenOchHavet** [reply]
Thanks for your helpful answers! :)
